text/0000-add-filename-info-to-io-error.md

```diff
@@ -0,0 +1,130 @@
+- Feature Name: `add-filename-info-to-io-error`
+- Start Date: 2021-02-11
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+- [Prior discussion](https://internals.rust-lang.org/t/add-filename-information-to-io-error/)
+- [Rust issue requesting for this change](https://github.com/rust-lang/rust/issues/44938)
+
+# Summary
+
+[summary]: #summary
+
+This RFC proposes to add filename information to `std::io:Error`. By doing so, `std::io` error messages will include which file the user is trying to access and making it easier for the user to notice where their code is failing.
+
+# Motivation
+
+[motivation]: #motivation
+
+Consider a simple program which reads a file which doesn't exist:
+
+```rust
+use std::fs::File;
+
+fn main() -> std::io::Result<()> {
+    let file = File::open("hello.txt")?;
+    println!("file: {:?}", file);
+    Ok(())
+}
+```
+
+This program emits the following error message:
+
+```txt
+Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
+```
+
+This error message doesn't tell the user which file doesn't exist and doesn't give the user a full breakdown of the error and what went wrong. As I've experienced and many others have, this error doesn't help a beginner Rust programmer to understand what went wrong and how to fix it. If the compiler emitted a full error and more information such as the name of the file which doesn't exist, this would be much more helpful.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+Suppose a user creates the same program as before:
+
+```rust
+use std::fs::File;
+
+fn main() -> std::io::Result<()> {
+    let file = File::open("hello.txt")?;
+    println!("file: {:?}", file);
+    Ok(())
+}
+```
+
+With this change, a much more descriptive and helpful error message will be emitted:
+
+```txt
+Error: Os { code: 2, kind: NotFound, message: "No such file or directory", file: "/Users/henryboisdequin/hello.txt" }
+```
+
+This error explains where the error is emitted and the cause of it. As you can see, this error is much more helpful. It points to where the problem is and the cause of it. Let's look at another case:
+
+```rust
+use std::fs::File;
+
+fn main() -> std::io::Result<()> {
+    let _ = File::create("/etc/protocols")?;
+
+    Ok(())
+}
+```
+
+This program returns the following error:
+
+```txt
+Error: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }
+```
+
+This program is not helpful at all, not displaying file information or how to fix this error. With this change, here is the projected error message:
+
+```
+Error: Os { code: 13, kind: PermissionDenied, message: "Permission denied", file: "Users/henryboisdequin/etc/protocols" }
```

Comment:
The path does not match the code above. Should be `/etc/protocols`.

Reply:
Thanks, it was a typo